crates/codegen/parser/runtime/src/mod_for_destination.rs

```diff
@@ -1,15 +1,48 @@
 #[macro_use]
 pub mod parser_support;
 
-pub mod cst;
-pub mod cursor;
 pub mod kinds;
 pub mod language;
 pub(crate) mod lexer;
 pub mod parse_error;
 pub mod parse_output;
-pub mod query;
-pub mod text_index;
+pub mod queries;
 
 #[cfg(feature = "slang_napi_interfaces")]
 pub mod napi_interface;
+
+mod metaslang_cst {
+    #[derive(Clone, Debug, PartialEq, Eq, serde::Serialize)] // But why?
```

Comment:
This is because default `#[derive(...)]` on a generic type implements only the trait with default bounds also implied for the generic types as well, i.e.

```rust
#[derive(Clone)] // expands to `impl<T: Clone> Clone for MyOption<T> { ... }` (notice the `T: Clone`)
struct MyOption<T>(Option<T>);
```

This assumes that the underlying data type uses this internally, however it's only used as a type container/marker.

One option is just derive it here and don't mind but a slightly more "correct" approach would be to implement the traits while skipping the bounds for the type marker, however this can be more noisy, see https://github.com/Xanewok/slang/commit/a034f8ad83d5806091706c93ac244862c80e7b79

Reply:
Let me know what you'd prefer; I'm fine with both approaches.

Reply:
I looked at the alternative - too noisy for me.